arroyo/processing/processor.py

```diff
@@ -111,10 +112,18 @@ def _close_strategy() -> None:
             logger.info("Waiting for %r to exit...", self.__processing_strategy)
 
             while True:
+                start_join = time.time()
+
                 try:
                     self.__processing_strategy.join(self.__join_timeout)
+                    self.__metrics_buffer.increment(
+                        ConsumerTiming.CONSUMER_JOIN_TIME, time.time() - start_join
+                    )
                     break
                 except InvalidMessage as e:
+                    self.__metrics_buffer.increment(
+                        ConsumerTiming.CONSUMER_JOIN_TIME, time.time() - start_join
+                    )
```

Comment:
should this be below line 127?

Reply:
if we do that, then CONSUMER_DLQ_TIME overlaps with CONSUMER_JOIN_TIME

Reply:
Yeah. I'm worried the join time is understated if we don't. Knowing the actual total join time is important here, not just the ratio of join time to other time (as we want to know how close to the 5 second mark - when we start getting those rebalance timeout errors - we are).

Reply:
could be CONSUMER_JOIN_DLQ_TIME?

Reply:
How about CONSUMER_SHUTDOWN_TIME is added as additional measurement? #215 already adds one that overlaps with join/poll timings

Reply:
Ignore that, separate CONSUMER_SHUTDOWN_TIME is better